### ISSUE TITLE
[FireMonkey] "Tab Scripts & CSS" sometimes does not show active scripts if url contains hash

### ISSUE THREAD

**Issue by Sanian-Creations** [erosman/support#576]
# Issue
Here's a table of different match patterns, whether the script is active on the page, and whether it is listed under "Tab Scripts & CSS" in the <img src="https://github.com/erosman/support/raw/master/image/firemonkey.png" height="15.5"> Toolbar Icon Pop-up.
|                                       | http://example.com/    | http://example.com/#whatever |
|---------------------------------------|:----------------------:|:----------------------------:|
| `@match http://example.com/`          | active, listed         | active, ***not listed (!)***     |
| `@match http://example.com/#whatever` | not active, not listed | not active, ***listed (!)***     |
| `@match http://example.com/*`         | active, listed         | active, listed               |

Obviously, whether a script is active and listed in the pop-up menu should always be the same.

I believe that all scripts run on the correct pages, just that the "Tab Scripts & CSS" list is wrong. You might argue that for the case with both a hash in the match *and* in the url, the script should be active even though it currently isn't, but I'm quite certain that hashes are not a part of the [Match Pattern spec](https://developer.chrome.com/docs/extensions/mv3/match_patterns/), and so match patterns with a hash should never work.

# How to reproduce
Add this style to FireMonkey, and then try out the combinations listed in the table above

```css
/*
==UserCSS==
@name             example
@match            http://example.com/
==/UserCSS==
*/

* { color: red; }
```

* If the script is active the text on the page should be red.
* If the script is listed you should see a `[1]` over the <img src="https://github.com/erosman/support/raw/master/image/firemonkey.png" height="15.5"> Toolbar Icon. (or open the pop-up to make sure if you have other active scripts)

# Cause and solution
It seems that there is a discrepancy between how scripts/styles are listed under "Tab Scripts & CSS" and how it is actually decided whether a script/style should run on a page. 

* To decide if a script should run, the hash is ignored, and then compared against the match.
* To decide if a script should show under "Tab Scripts & CSS", the hash is NOT ignored, and then compared against the match. *This* is the problem, and should be made to ignore hashes also.

**Comment by erosman** [comment]
The script injection and the popup display are completely difference processes:

- Script injection is done via Firefox API which follows [Match Patterns](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/match_patterns)
- Matching in Popup display is done manually by FireMonkey, using Regular Expression to match scripts to the Tab URL

The popup display process should remove the hash i.e.

https://github.com/erosman/support/blob/599da969c3cd143235f6f5765b587198ccf71dd5/content/match.js#L16

https://github.com/erosman/support/blob/599da969c3cd143235f6f5765b587198ccf71dd5/content/match.js#L37-L39


I will check if there is an oversight.

**Comment by erosman** [comment]
Found the reason. Fixed for v2.72
